bip-0352.mediawiki

```diff
@@ -0,0 +1,475 @@
+
+The one exception is script path spends that use NUMS point ''H'' as their internal key (where ''H = lift_x(0x50929b74c1a04954b78b4b6035e97a5e078a5a0f28ec96d547bfee9ace803ac0)'' which is constructed by taking the hash of the standard uncompressed encoding of the secp256k1 base point ''G'' as X coordinate, see [https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#constructing-and-spending-taproot-outputs BIP341: Constructing and spending Taproot outputs] for more details), in which case the output will be skipped for the purposes of shared secret derivation<ref name="why_ignore_h">'''Why skip outputs with H as the internal taproot key?''' If use cases get popularized where the taproot key path cannot be used, these outputs can still be included without getting in the way of making a silent payment, provided they specifically use H as their internal taproot key.</ref>.
+
+''' P2WPKH '''
+
+The sender performs the tweak using the private key for the output and the receiver obtains the public key from the ''witness''.
+
+''' P2SH-P2WPKH '''
+
+The sender performs the tweak using the private key for the nested ''P2WPKH'' output and the receiver obtains the public key from the ''witness''.
+
+''' P2PKH '''
+
+The sender performs the tweak using the private key for the output and SHOULD sign using the standard script template:
+
+
+    scriptSig: <Signature> <Public Key>
+
+
+The receiver obtains the public key from the ''scriptSig''. The receiver MUST parse the ''scriptSig'' for the public key, even if the ''scriptSig'' is non-standard (e.g. <code><dummy> OP_DROP <Signature> <Public Key></code>). This is to address the [https://en.bitcoin.it/wiki/Transaction_malleability third-party malleability of ''P2PKH'' ''scriptSigs''].
+
+''' P2PK '''
+
+The sender performs the tweak using the private key for the output and the receiver obtains the public key from the ''scriptPubKey''.
+
+=== Outpoint hash ===
+
+The sender and receiver MUST calculate an outpoints hash for the transaction in the following manner:
+
+* Let ''A = A<sub>0</sub> + A<sub>1</sub> + ... + A<sub>n</sub>'', where each ''A<sub>i</sub>'' is the public key of an input from the ''[[#inputs-for-shared-secret-derivation|Inputs For Shared Secret Derivation]]'' list
+* Let ''outpoint = outpoint<sub>0</sub>'',  where ''outpoint<sub>0</sub>'' is the lowest outpoint lexicographically by txid and vout used in the transaction<ref name="why_lowest_outpoint"></ref>
+* Let ''outpoint_hash = hash(outpoint || A)''
```

Comment:
1. What's the reason for including the sum of the public keys in the hash?
2. Since this now includes both an outpoint and input public keys, I think a better name would be something like "input nonce".

Reply:
> What's the reason for including the sum of the public keys in the hash?

The discussion that led to including the sum of the public keys is here. I also wrote a footnote for the overview section explaining why, but realize I forgot to link it here. I'll add that.

>  I think a better name would be something like "input nonce".

I like this. In the reference python, I renamed it to deterministic_nonce. Maybe we can do `deterministic_input_nonce`?

Reply:
For me "input nonce" already implies it's deterministic, so I'd prefer either that or "deterministic nonce". Don't care all that much though as long as it's not "outpoint hash".

Reply:
I think input nonce makes the most sense in this context.

Reply:
At the risk of bike shedding, I really don't like the term "nonce". Yeah, the original  meaning is "number used once" and this is not wrong here, but when people hear "nonce" in the context of Bitcoin / signatures, they'll think about the EC points in ECDSA or Schnorr sigs (or their dlogs perhaps). 

But agreed, "outpoint hash" is also misleading now after inclusion of A. Perhaps just "input hash"? Or "utxo_hash"? The latter is closer to "outpoint hash" but is probably less confusion because an "UTXO" is not a precisely defined string (as opposed to an outpoint).

Reply:
Also, there are currently still occurrences of the term "outpoint hash" in the doc, e.g., in the header.

Reply:
"Input hash" sounds good to me if we want to avoid "nonce".

Reply:
Updated to input hash